client/gutenberg/extensions/giphy/edit.js

```diff
@@ -16,6 +16,8 @@ class GiphyEdit extends Component {
 
 	state = {
 		focus: false,
+		results: null,
+		thumbnails: null,
```

Comment:
`thumbnails` are `results` without the selectedImage, right? Rather than duplicating computed information in state, I'd propose that we tweak this a bit:

- Store the `giphyId` attribute rather than the `giphyUrl`
- Keep the results in state
- Produce a useable URL from the `giphyId` on render
- Filter the results (`result.id !== this.props.attributes.giphyId`) to render the alternatives.

Reply:
Sneaky, sneaky, bringing back your `giphyId` proposal via a back door!

Here's a re-work that adopts this approach, without going all the way on replacing `giphyUrl` with `giphyId`. How's this suit you?

Reply:
> Sneaky, sneaky, bringing back your `giphyId` proposal via a back door!

✋ Guilty as charged 🙂 Either I won't let that go 🐶 🍖  or I have stellar intuition ✨ 🤷‍♂️ 

> Here's a re-work that adopts this approach, without going all the way on replacing `giphyUrl` with `giphyId`. How's this suit you?
> [bd37d65](https://github.com/Automattic/wp-calypso/commit/bd37d6565931a542697d1d3d9f488a28fecae815)

👍 That should be good, I'll have a look and test. The goal is to remove the redundant state and it sounds like we got it.